File: 5_Make_Control_Plots/utils/makeROOTFiles.cpp
//cpp
#include <vector>
#include <string>
#include <iostream>
#include <fstream>
#include <sstream>
#include <typeinfo>
//#include <boost/lexical_cast.hpp>


using namespace std;

void makeROOTFiles() {

	fstream fs;
	fs.open("tt_2l2nu.csv",ios::in);
	string line;

	std::vector<double> jet1_mass,jet1_pt,jet1_eta,jet1_phi,jet1_btag; 
	std::vector<double> jet2_mass,jet2_pt,jet2_eta,jet2_phi,jet2_btag; 
	std::vector<double> lep1_mass,lep1_pt,lep1_eta,lep1_phi,lep1_charge; 
	std::vector<double> lep2_mass,lep2_pt,lep2_eta,lep2_phi,lep2_charge;

	std::vector<double> higgs1_mass,higgs1_pt,higgs1_eta,higgs1_phi;
	std::vector<double> higgs2_mass,higgs2_pt,higgs2_eta,higgs2_phi;
	std::vector<double> hhiggs_mass,hhiggs_pt,hhiggs_eta,hhiggs_phi;
	std::vector<double> deltar_ll,deltar_jj;

	int cc = 0;
	//while(!fs.eof()) {
	//while(cc < 2051) {
	//while(cc < 4001) {
	while(cc < 10000) {

		getline(fs,line);

		istringstream ss(line);
		string buffer;
		std::vector<string> data;

		while(getline(ss,buffer,',')) { data.push_back(buffer); }

		//if (cc > 0) { cout << stod(data.at(0)) << endl; }
		if (cc > 0) {

			double jet1_mass_d,jet1_pt_d,jet1_eta_d,jet1_phi_d,jet1_btag_d;
			double jet2_mass_d,jet2_pt_d,jet2_eta_d,jet2_phi_d,jet2_btag_d;
			double lep1_mass_d,lep1_pt_d,lep1_eta_d,lep1_phi_d;
			double lep2_mass_d,lep2_pt_d,lep2_eta_d,lep2_phi_d;

			double h1_mass_d,h1_pt_d,h1_eta_d,h1_phi_d;
			double h2_mass_d,h2_pt_d,h2_eta_d,h2_phi_d;
			double hh_mass_d,hh_pt_d,hh_eta_d,hh_phi_d;
			double dr_ll_d,dr_jj_d;

			int lep1_charge_i,lep2_charge_i;

			jet1_mass_d = stod(data.at(0));
			jet1_pt_d = stod(data.at(1));
			jet1_eta_d = stod(data.at(2));
			jet1_phi_d = stod(data.at(3));
			jet1_btag_d = stod(data.at(4));

			jet2_mass_d = stod(data.at(5));
			jet2_pt_d = stod(data.at(6));
			jet2_eta_d = stod(data.at(7));
			jet2_phi_d = stod(data.at(8));
			jet2_btag_d = stod(data.at(9));

			lep1_mass_d = stod(data.at(10));
			lep1_pt_d = stod(data.at(11));
			lep1_eta_d = stod(data.at(12));
			lep1_phi_d = stod(data.at(13));
			lep1_charge_i = stoi(data.at(14));

			lep2_mass_d = stod(data.at(15));
			lep2_pt_d = stod(data.at(16));
			lep2_eta_d = stod(data.at(17));
			lep2_phi_d = stod(data.at(18));
			lep2_charge_i = stoi(data.at(19));

			dr_jj_d = stod(data.at(20));
			dr_ll_d = stod(data.at(21));

			h1_mass_d = stod(data.at(22));
			h1_pt_d = stod(data.at(23));
			h1_eta_d = stod(data.at(24));
			h1_phi_d = stod(data.at(25));

			h2_mass_d = stod(data.at(26));
			h2_pt_d = stod(data.at(27));
			h2_eta_d = stod(data.at(28));
			h2_phi_d = stod(data.at(29));

			hh_mass_d = stod(data.at(30));
			hh_pt_d = stod(data.at(31));
			hh_eta_d = stod(data.at(32));
			hh_phi_d = stod(data.at(33));

			jet1_mass.push_back(jet1_mass_d);
			jet1_pt.push_back(jet1_pt_d);
			jet1_eta.push_back(jet1_eta_d);
			jet1_phi.push_back(jet1_phi_d);
			jet1_btag.push_back(jet1_btag_d);

			jet2_mass.push_back(jet2_mass_d);
			jet2_pt.push_back(jet2_pt_d);
			jet2_eta.push_back(jet2_eta_d);
			jet2_phi.push_back(jet2_phi_d);
			jet2_btag.push_back(jet2_btag_d);

			lep1_mass.push_back(lep1_mass_d);
			lep1_pt.push_back(lep1_pt_d);
			lep1_eta.push_back(lep1_eta_d);
			lep1_phi.push_back(lep1_phi_d);
			lep1_charge.push_back(lep1_charge_i);

			lep2_mass.push_back(lep2_mass_d);
			lep2_pt.push_back(lep2_pt_d);
			lep2_eta.push_back(lep2_eta_d);
			lep2_phi.push_back(lep2_phi_d);
			lep2_charge.push_back(lep2_charge_i);

			deltar_jj.push_back(dr_jj_d);
			deltar_ll.push_back(dr_ll_d);

			higgs1_mass.push_back(h1_mass_d);
			higgs1_pt.push_back(h1_pt_d);
			higgs1_eta.push_back(h1_eta_d);
			higgs1_phi.push_back(h1_phi_d);

			higgs2_mass.push_back(h2_mass_d);
			higgs2_pt.push_back(h2_pt_d);
			higgs2_eta.push_back(h2_eta_d);
			higgs2_phi.push_back(h2_phi_d);

			hhiggs_mass.push_back(hh_mass_d);
			hhiggs_pt.push_back(hh_pt_d);
			hhiggs_eta.push_back(hh_eta_d);
			hhiggs_phi.push_back(hh_phi_d);
		}
		data.clear();
		cc++;
	}

//	for (vector<double>::iterator it=h1_mass.begin()+1; it!=h1_mass.end(); it++)
//	{ cout << *it << endl; }

	TFile* f = TFile::Open("tmva_test.root","recreate");
	TTree* test = new TTree("test","test tree");
	for (int i=0; i<cc-1; i++) {

		double j1_mass,j1_pt,j1_eta,j1_phi,j1_btag;
		double j2_mass,j2_pt,j2_eta,j2_phi,j2_btag;
		double l1_mass,l1_pt,l1_eta,l1_phi;
		double l2_mass,l2_pt,l2_eta,l2_phi;
		double dr_ll,dr_jj;
		double h1_mass,h1_pt,h1_eta,h1_phi;
		double h2_mass,h2_pt,h2_eta,h2_phi;
		double hh_mass,hh_pt,hh_eta,hh_phi;
		int l1_charge,l2_charge;

		test->Branch("J1_mass", &j1_mass, "j1_mass/D");
		test->Branch("J1_pt", &j1_pt, "j1_pt/D");
		test->Branch("J1_eta", &j1_eta, "j1_eta/D");
		test->Branch("J1_phi", &j1_phi, "j1_phi/D");
		test->Branch("J1_btag", &j1_btag, "j1_btag/D");

		test->Branch("J2_mass", &j2_mass, "j2_mass/D");
                test->Branch("J2_pt", &j2_pt, "j2_pt/D");
                test->Branch("J2_eta", &j2_eta, "j2_eta/D");
                test->Branch("J2_phi", &j2_phi, "j2_phi/D");
                test->Branch("J2_btag", &j2_btag, "j2_btag/D");

		test->Branch("L1_mass", &l1_mass, "l1_mass/D");
                test->Branch("L1_pt", &l1_pt, "l1_pt/D");
                test->Branch("L1_eta", &l1_eta, "l1_eta/D");
                test->Branch("L1_phi", &l1_phi, "l1_phi/D");
                test->Branch("L1_charge", &l1_charge, "l1_charge/I");

                test->Branch("L2_mass", &l2_mass, "l2_mass/D");
                test->Branch("L2_pt", &l2_pt, "l2_pt/D");
                test->Branch("L2_eta", &l2_eta, "l2_eta/D");
                test->Branch("L2_phi", &l2_phi, "l2_phi/D");
                test->Branch("L2_charge", &l2_charge, "l2_charge/I");

		test->Branch("DeltaR_ll", &dr_ll, "dr_ll/D");
		test->Branch("DeltaR_jj", &dr_jj, "dr_jj/D");

		test->Branch("H1_mass", &h1_mass, "h1_mass/D");
		test->Branch("H1_pt", &h1_pt, "h1_pt/D");
		test->Branch("H1_eta", &h1_eta, "h1_eta/D");
		test->Branch("H1_phi", &h1_phi, "h1_phi/D");

		test->Branch("H2_mass", &h2_mass, "h2_mass/D");
                test->Branch("H2_pt", &h2_pt, "h2_pt/D");
                test->Branch("H2_eta", &h2_eta, "h2_eta/D");
                test->Branch("H2_phi", &h2_phi, "h2_phi/D");

		test->Branch("HH_mass", &hh_mass, "hh_mass/D");
                test->Branch("HH_pt", &hh_pt, "hh_pt/D");
                test->Branch("HH_eta", &hh_eta, "hh_eta/D");
                test->Branch("HH_phi", &hh_phi, "hh_phi/D");

		j1_mass = jet1_mass.at(i);
		j1_pt   = jet1_pt.at(i);
		j1_eta  = jet1_eta.at(i);
		j1_phi  = jet1_phi.at(i);
		j1_btag = jet1_btag.at(i);

		j2_mass = jet2_mass.at(i);
                j2_pt   = jet2_pt.at(i);
                j2_eta  = jet2_eta.at(i);
                j2_phi  = jet2_phi.at(i);
                j2_btag = jet2_btag.at(i);

		l1_mass = lep1_mass.at(i);
                l1_pt   = lep1_pt.at(i);
                l1_eta  = lep1_eta.at(i);
                l1_phi  = lep1_phi.at(i);
                l1_charge = lep1_charge.at(i);

                l2_mass = lep2_mass.at(i);
                l2_pt   = lep2_pt.at(i);
                l2_eta  = lep2_eta.at(i);
                l2_phi  = lep2_phi.at(i);
                l2_charge = lep2_charge.at(i);

		dr_ll   = deltar_ll.at(i);
		dr_jj   = deltar_jj.at(i);

		h1_mass = higgs1_mass.at(i);
		h1_pt   = higgs1_pt.at(i);
		h1_eta  = higgs1_eta.at(i);
		h1_phi  = higgs1_phi.at(i);

		h2_mass = higgs2_mass.at(i);
                h2_pt   = higgs2_pt.at(i);
                h2_eta  = higgs2_eta.at(i);
                h2_phi  = higgs2_phi.at(i);

		hh_mass = hhiggs_mass.at(i);
                hh_pt   = hhiggs_pt.at(i);
                hh_eta  = hhiggs_eta.at(i);
                hh_phi  = hhiggs_phi.at(i);

		test->Fill();
	}
	f->cd();
	test->Write();
	f->Close();

	std::cout << "Successfully generated root file..." << std::endl;
}

File: 5_Make_Control_Plots/2016/WORK_2016/CSV/makeROOTFiles.py
#python
# -*- coding: utf-8 -*-
from ROOT import *
from glob import glob
import numpy as np 
import math
import csv 
import sys
import ast

def main():


	flist = glob('*.csv')
	for fname in flist:

		dr = fname[:-4]
		f = open('%s.csv'%(dr),'r')
		readline = csv.reader(f)

		jet1_mass,jet1_pt,jet1_eta,jet1_phi,jet1_btag = [],[],[],[],[]
		jet2_mass,jet2_pt,jet2_eta,jet2_phi,jet2_btag = [],[],[],[],[]
		lep1_mass,lep1_pt,lep1_eta,lep1_phi,lep1_charge = [],[],[],[],[]
		lep2_mass,lep2_pt,lep2_eta,lep2_phi,lep2_charge = [],[],[],[],[]
		H1_mass,H1_pt,H1_eta,H1_phi = [],[],[],[]
		H2_mass,H2_pt,H2_eta,H2_phi = [],[],[],[]
		HH_mass,HH_pt,HH_eta,HH_phi = [],[],[],[]
		dr_ll,dr_jj,genWeight = [],[],[]

		ll = 0
		for line in readline:

			jet1_mass.append(float(line[0]))
			jet1_pt.append(float(line[1]))
			jet1_eta.append(float(line[2]))
			jet1_phi.append(float(line[3]))
			jet1_btag.append(float(line[4]))

			jet2_mass.append(float(line[5]))
			jet2_pt.append(float(line[6]))
			jet2_eta.append(float(line[7]))
			jet2_phi.append(float(line[8]))
			jet2_btag.append(float(line[9]))

			lep1_mass.append(float(line[10]))
			lep1_pt.append(float(line[11]))
			lep1_eta.append(float(line[12]))
			lep1_phi.append(float(line[13]))
			lep1_charge.append(float(line[14]))

			lep2_mass.append(float(line[15]))
			lep2_pt.append(float(line[16]))
			lep2_eta.append(float(line[17]))
			lep2_phi.append(float(line[18]))
			lep2_charge.append(float(line[19]))

			dr_jj.append(float(line[20]))
			dr_ll.append(float(line[21]))

			H1_mass.append(float(line[22]))
			H1_pt.append(float(line[23]))
			H1_eta.append(float(line[24]))
			H1_phi.append(float(line[25]))

			H2_mass.append(float(line[26]))
			H2_pt.append(float(line[27]))
			H2_eta.append(float(line[28]))
			H2_phi.append(float(line[29]))

			HH_mass.append(float(line[30]))
			HH_pt.append(float(line[31]))
			HH_eta.append(float(line[32]))
			HH_phi.append(float(line[33]))

			genWeight.append(float(line[-1]))
			ll += 1

		f.close()


		h_J1_mass = TH1D("J1_mass", "Mass Distribution of Leading Jet", 20, 0, 200)
		h_J1_mass.GetXaxis().SetTitle("Invariant mass [GeV]")
		h_J1_pt   = TH1D("J1_pt", "Transverse Momentum Distribution of Leading Jet", 20, 0, 200)
		h_J1_pt.GetXaxis().SetTitle("Transverse momentum [GeV]")
		h_J1_eta  = TH1D("J1_eta", "Eta Distribution of Leading Jet", 20, -2.4, 2.4)
		h_J1_eta.GetXaxis().SetTitle("Eta")
		h_J1_phi  = TH1D("J1_phi", "Phi Distribution of Leading Jet", 20, -3.14, 3.14)
		h_J1_phi.GetXaxis().SetTitle("Phi")
		h_J1_btag  = TH1D("J1_btag", "btagDeepFlavB Distribution of Leading Jet", 20, 0, 1.01)
		h_J1_btag.GetXaxis().SetTitle("btagDeepFlavB")

		h_J2_mass = TH1D("J2_mass", "Mass Distribution of Sub-leading Jet", 20, 0, 200)
		h_J2_mass.GetXaxis().SetTitle("Invariant mass [GeV]")
		h_J2_pt   = TH1D("J2_pt", "Transverse Momentum Distribution of Sub-leading Jet", 20, 0, 200)
		h_J2_pt.GetXaxis().SetTitle("Transverse momentum [GeV]")
		h_J2_eta  = TH1D("J2_eta", "Eta Distribution of Sub-leading Jet", 20, -2.4, 2.4)
		h_J2_eta.GetXaxis().SetTitle("Eta")
		h_J2_phi  = TH1D("J2_phi", "Phi Distribution of Sub-leading Jet", 20, -3.14, 3.14)
		h_J2_phi.GetXaxis().SetTitle("Phi")
		h_J2_btag  = TH1D("J2_btag", "btagDeepFlavB Distribution of Sub-leading Jet", 20, 0, 1.01)
		h_J2_btag.GetXaxis().SetTitle("btagDeepFlavB")

		h_L1_mass = TH1D("L1_mass", "Mass Distribution of Leading Lepton", 20, 0, 200)
		h_L1_mass.GetXaxis().SetTitle("Invariant mass [GeV]")
		h_L1_pt   = TH1D("L1_pt", "Transverse Momentum Distribution of Leading Lepton", 20, 0, 200)
		h_L1_pt.GetXaxis().SetTitle("Transverse momentum [GeV]")
		h_L1_eta  = TH1D("L1_eta", "Eta Distribution of Leading Lepton", 20, -2.4, 2.4)
		h_L1_eta.GetXaxis().SetTitle("Eta")
		h_L1_phi  = TH1D("L1_phi", "Phi Distribution of Leading Lepton", 20, -3.14, 3.14)
		h_L1_phi.GetXaxis().SetTitle("Phi")
		h_L1_charge = TH1D("L1_charge", "Charge Distribution of Leading Lepton", 20, -1, 1.01)
		h_L1_charge.GetXaxis().SetTitle("Charge")

		h_L2_mass = TH1D("L2_mass", "Mass Distribution of Sub-leading Lepton", 20, 0, 200)
		h_L2_mass.GetXaxis().SetTitle("Invariant mass [GeV]")
		h_L2_pt   = TH1D("L2_pt", "Transverse Momentum Distribution of Sub-leading Lepton", 20, 0, 200)
		h_L2_pt.GetXaxis().SetTitle("Transverse momentum [GeV]")
		h_L2_eta  = TH1D("L2_eta", "Eta Distribution of Sub-leading Lepton", 20, -2.4, 2.4)
		h_L2_eta.GetXaxis().SetTitle("Eta")
		h_L2_phi  = TH1D("L2_phi", "Phi Distribution of Sub-leading Lepton", 20, -3.14, 3.14)
		h_L2_phi.GetXaxis().SetTitle("Phi")
		h_L2_charge = TH1D("L2_charge", "Charge Distribution of Sub-leading Lepton", 20, -1, 1.01)
		h_L2_charge.GetXaxis().SetTitle("Charge")

		h_H1_mass = TH1D("H1_mass", "Mass Distribution of H1 <- MuMu", 20, 0, 400)
		h_H1_mass.GetXaxis().SetTitle("Invariant mass [GeV]")
		h_H1_pt   = TH1D("H1_pt", "Transverse Momentum Distribution of H1 <- MuMu", 20, 0, 400)
		h_H1_pt.GetXaxis().SetTitle("Transverse momentum [GeV]")
		h_H1_eta  = TH1D("H1_eta", "Eta Distribution of H1 <- MuMu", 20, -2.4, 2.4)
		h_H1_eta.GetXaxis().SetTitle("Eta")
		h_H1_phi  = TH1D("H1_phi", "Phi Distribution of H1 <- MuMu", 20, -3.14, 3.14)
		h_H1_phi.GetXaxis().SetTitle("Phi")

		h_H2_mass = TH1D("H2_mass", "Mass Distribution of H2 <- bb", 20, 0, 400)
		h_H2_mass.GetXaxis().SetTitle("Invariant mass [GeV]")
		h_H2_pt   = TH1D("H2_pt", "Transverse Momentum Distribution of H2 <- bb", 20, 0, 400)
		h_H2_pt.GetXaxis().SetTitle("Transverse momentum [GeV]")
		h_H2_eta  = TH1D("H2_eta", "Eta Distribution of H2 <- bb", 20, -2.4, 2.4)
		h_H2_eta.GetXaxis().SetTitle("Eta")
		h_H2_phi  = TH1D("H2_phi", "Phi Distribution of H2 <- bb", 20, -3.14, 3.14)
		h_H2_phi.GetXaxis().SetTitle("Phi")

		h_HH_mass = TH1D("HH_mass", "Mass Distribution of HH", 20, 0, 800)
		h_HH_mass.GetXaxis().SetTitle("Invariant mass [GeV]")
		h_HH_pt   = TH1D("HH_pt", "Transverse Momentum Distribution of HH", 20, 0, 800)
		h_HH_pt.GetXaxis().SetTitle("Transverse momentum [GeV]")
		h_HH_eta  = TH1D("HH_eta", "Eta Distribution of HH", 20, -2.4, 2.4)
		h_HH_eta.GetXaxis().SetTitle("Eta")
		h_HH_phi  = TH1D("HH_phi", "Phi Distribution of HH", 20, -3.14, 3.14)
		h_HH_phi.GetXaxis().SetTitle("Phi")

		h_dr_ll = TH1D("DeltaR_ll", "DeltaR Distribution between Selected Leptons", 20, 0, 10)
		h_dr_ll.GetXaxis().SetTitle("DeltaR")
		h_dr_jj = TH1D("DeltaR_jj", "DeltaR Distribution between Selected Jets", 20, 0, 10)
		h_dr_jj.GetXaxis().SetTitle("DeltaR")


		genweight = 0
		for i in range(len(jet1_mass)):

			if (jet1_mass[i] >= h_J1_mass.GetBinCenter(20)): jet1_mass[i] = h_J1_mass.GetBinCenter(20)
			h_J1_mass.Fill(jet1_mass[i],genWeight[i])
			if (jet1_pt[i] >= h_J1_pt.GetBinCenter(20)): jet1_pt[i] = h_J1_pt.GetBinCenter(20)
			h_J1_pt.Fill(jet1_pt[i],genWeight[i])
			h_J1_eta.Fill(jet1_eta[i],genWeight[i])
			h_J1_phi.Fill(jet1_phi[i],genWeight[i])
			h_J1_btag.Fill(jet1_btag[i],genWeight[i])

			if (jet2_mass[i] >= h_J2_mass.GetBinCenter(20)): jet2_mass[i] = h_J2_mass.GetBinCenter(20)
                        h_J2_mass.Fill(jet2_mass[i],genWeight[i])
                        if (jet2_pt[i] >= h_J2_pt.GetBinCenter(20)): jet2_pt[i] = h_J2_pt.GetBinCenter(20)
                        h_J2_pt.Fill(jet2_pt[i],genWeight[i])
                        h_J2_eta.Fill(jet2_eta[i],genWeight[i])
                        h_J2_phi.Fill(jet2_phi[i],genWeight[i])
                        h_J2_btag.Fill(jet2_btag[i],genWeight[i])

			if (lep1_mass[i] >= h_L1_mass.GetBinCenter(20)): lep1_mass[i] = h_L1_mass.GetBinCenter(20)
                        h_L1_mass.Fill(lep1_mass[i],genWeight[i])
                        if (lep1_pt[i] >= h_L1_pt.GetBinCenter(20)): lep1_pt[i] = h_L1_pt.GetBinCenter(20)
                        h_L1_pt.Fill(lep1_pt[i],genWeight[i])
                        h_L1_eta.Fill(lep1_eta[i],genWeight[i])
                        h_L1_phi.Fill(lep1_phi[i],genWeight[i])
                        h_L1_charge.Fill(lep1_charge[i],genWeight[i])

			if (lep2_mass[i] >= h_L2_mass.GetBinCenter(20)): lep2_mass[i] = h_L2_mass.GetBinCenter(20)
                        h_L2_mass.Fill(lep2_mass[i],genWeight[i])
                        if (lep2_pt[i] >= h_L2_pt.GetBinCenter(20)): lep2_pt[i] = h_L2_pt.GetBinCenter(20)
                        h_L2_pt.Fill(lep2_pt[i],genWeight[i])
                        h_L2_eta.Fill(lep2_eta[i],genWeight[i])
                        h_L2_phi.Fill(lep2_phi[i],genWeight[i])
                        h_L2_charge.Fill(lep2_charge[i],genWeight[i])

                        if (dr_ll[i] >= h_dr_ll.GetBinCenter(20)): dr_ll[i] = h_dr_ll.GetBinCenter(20)
			h_dr_ll.Fill(dr_ll[i],genWeight[i])
                        if (dr_jj[i] >= h_dr_jj.GetBinCenter(20)): dr_jj[i] = h_dr_jj.GetBinCenter(20)
			h_dr_jj.Fill(dr_jj[i],genWeight[i])

			if (H1_mass[i] >= h_H1_mass.GetBinCenter(20)): H1_mass[i] = h_H1_mass.GetBinCenter(20)
                        h_H1_mass.Fill(H1_mass[i],genWeight[i])
                        if (H1_pt[i] >= h_H1_pt.GetBinCenter(20)): H1_pt[i] = h_H1_pt.GetBinCenter(20)
                        h_H1_pt.Fill(H1_pt[i],genWeight[i])
                        h_H1_eta.Fill(H1_eta[i],genWeight[i])
                        h_H1_phi.Fill(H1_phi[i],genWeight[i])

			if (H2_mass[i] >= h_H2_mass.GetBinCenter(20)): H2_mass[i] = h_H2_mass.GetBinCenter(20)
                        h_H2_mass.Fill(H2_mass[i],genWeight[i])
                        if (H2_pt[i] >= h_H2_pt.GetBinCenter(20)): H2_pt[i] = h_H2_pt.GetBinCenter(20)
                        h_H2_pt.Fill(H2_pt[i],genWeight[i])
                        h_H2_eta.Fill(H2_eta[i],genWeight[i])
                        h_H2_phi.Fill(H2_phi[i],genWeight[i])

			if (HH_mass[i] >= h_HH_mass.GetBinCenter(20)): HH_mass[i] = h_HH_mass.GetBinCenter(20)
                        h_HH_mass.Fill(HH_mass[i],genWeight[i])
                        if (HH_pt[i] >= h_HH_pt.GetBinCenter(20)): HH_pt[i] = h_HH_pt.GetBinCenter(20)
                        h_HH_pt.Fill(HH_pt[i],genWeight[i])
                        h_HH_eta.Fill(HH_eta[i],genWeight[i])
                        h_HH_phi.Fill(HH_phi[i],genWeight[i])

			genweight += genWeight[i]


		ff = TFile("./%s.root"%(dr),"recreate")
		ff.cd()
		gDirectory.mkdir(dr)
		ff.cd(dr)
		
		h_J1_mass.Write()
		h_J1_pt.Write()
		h_J1_eta.Write()
		h_J1_phi.Write()
		h_J1_btag.Write()

		h_J2_mass.Write()
		h_J2_pt.Write()
		h_J2_eta.Write()
		h_J2_phi.Write()
		h_J2_btag.Write()

		h_L1_mass.Write()
		h_L1_pt.Write()
		h_L1_eta.Write()
		h_L1_phi.Write()
		h_L1_charge.Write()

		h_L2_mass.Write()
		h_L2_pt.Write()
		h_L2_eta.Write()
		h_L2_phi.Write()
		h_L2_charge.Write()

		h_dr_ll.Write()
		h_dr_jj.Write()

		h_H1_mass.Write()
		h_H1_pt.Write()
		h_H1_eta.Write()
		h_H1_phi.Write()

		h_H2_mass.Write()
		h_H2_pt.Write()
		h_H2_eta.Write()
		h_H2_phi.Write()

		h_HH_mass.Write()
		h_HH_pt.Write()
		h_HH_eta.Write()
		h_HH_phi.Write()

		ff.Close()
		print("Processed %d events..."%(ll))
		print("Successfully generated root file...")


if __name__ == "__main__":
        main()
